backend/typescript/prisma/schema.prisma

```diff
@@ -0,0 +1,77 @@
+model Task {
+    id                      Int @id @default(autoincrement())
+    category                Categories 
+    title                   String   
+    status                  Statuses
+    description             String
+    assignee                Resident @relation(fields: [assignee_id], references: [id])
+    assignee_id             Int
+    assigner                Staff @relation(fields: [assigner_id], references: [id])
+    assigner_id             Int
+    credit_value            Float
+    start_date              DateTime
+    end_date                DateTime?
+    comments                String
+    recurrence_frequency    String?
+    // recurrence_days String // store days a comma separted string to be parsed for usage
+    warnings                Warning[]
+}
+
+enum Categories {
+    HOUSEHOLD
+    CHORES
+    FINANCE
+    // other categories
+}
+
+enum Statuses {
+    PENDING_APPROVAL
+    INCOMPLETE
+    COMPLETE
+    EXCUSED
+}
+
+model Warning {
+    resident_id             Int @id @default(autoincrement())
+    title                   String
+    description             String
+    assigner                Staff @relation(fields: [assigner_id], references: [id])
+    assigner_id             Int
+    date_issued             DateTime
+    related_task            Task? @relation(fields: [related_task_id], references: [id])
+    related_task_id         Int?
+}
+
+model Staff {
+	id     					Int @id @default(autoincrement())
+	role   					StaffRoles
+	first_name 				String
+	last_name 				String
+	email 					String
+	phone_number 			String?
+	display_name 			String
+	profile_picture_link 	String?
+    tasks                   Task[]
+    warnings                Warning[]
```

Comment:
Should we rename these to `tasks_assigned` and `warnings_assigned` so that is clear that there aren't their tasks and warnings?